### PR TITLE
Refact to reduce compile times

### PR DESCRIPTION
**Pull request by szabolcs-instructure**
[ignore-commit-lint]

### CODE REVIEW

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Build failed, skipping coverage check
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  Please add a reference to a jira ticket. For example: `refs: MBL-10023`
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  Please add which apps this change affects. Example: `affects: Teacher, Student` or `affects: none`
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  Please add a release note. If no release note is wanted, use `none`. Example: `release note: Fixed a bug that prevented users from enjoying the app.`
  </td>
    </tr>
  </tbody>
</table>



<details><summary>:x: Swift lint</summary>


    ❌ /Users/vagrant/git/Core/Core/Assignments/AssignmentEditor/AssignmentAssigneePicker.swift:121:1: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
    /Users/vagrant/git/Core/Core/Assignments/AssignmentEditor/AssignmentEditorView.swift:102:1: error: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
    /Users/vagrant/git/Core/Core/Assignments/AssignmentEditor/AssignmentEditorView.swift:111:1: error: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
    ❌ /Users/vagrant/git/Core/Core/Assignments/AssignmentEditor/AssignmentEditorView.swift:132:1: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
    /Users/vagrant/git/Core/Core/Assignments/AssignmentEditor/AssignmentEditorView.swift:172:1: error: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
    /Users/vagrant/git/Core/Core/Dashboard/K5/View/Grades/K5GradesView.swift:88:1: error: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
    ❌ /Users/vagrant/git/Core/Core/Dashboard/K5/View/Grades/K5GradesView.swift:105:1: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
    /Users/vagrant/git/rn/Teacher/ios/Teacher/SpeedGrader/SubmissionGrades.swift:168:1: error: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
    /Users/vagrant/git/rn/Teacher/ios/Teacher/SpeedGrader/RubricAssessor.swift:122:1: error: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
    ❌ /Users/vagrant/git/rn/Teacher/ios/Teacher/SpeedGrader/RubricAssessor.swift:123:1: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
    /Users/vagrant/git/rn/Teacher/ios/Teacher/SpeedGrader/RubricAssessor.swift:130:1: error: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
    /Users/vagrant/git/rn/Teacher/ios/Teacher/SpeedGrader/RubricAssessor.swift:137:1: error: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
    ❌ /Users/vagrant/git/rn/Teacher/ios/Teacher/SpeedGrader/RubricAssessor.swift:161:1: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
    /Users/vagrant/git/rn/Teacher/ios/Teacher/SpeedGrader/RubricAssessor.swift:123:1: error: Vertical Whitespace Violation: Limit vertical whitespace to a single empty line. Currently 2. (vertical_whitespace)
    /Users/vagrant/git/Core/Core/API/APIError.swift:74:31: error: Colon Spacing Violation: Colons should be next to the identifier when specifying a type and next to the key in dictionary literals. (colon)
    ❌ /Users/vagrant/git/Core/Core/API/APIError.swift:74:42: Colon Spacing Violation: Colons should be next to the identifier when specifying a type and next to the key in dictionary literals. (colon)
    /Users/vagrant/git/Core/Core/API/APIError.swift:74:52: error: Colon Spacing Violation: Colons should be next to the identifier when specifying a type and next to the key in dictionary literals. (colon)
    /Users/vagrant/git/Core/Core/API/APIError.swift:75:44: error: Colon Spacing Violation: Colons should be next to the identifier when specifying a type and next to the key in dictionary literals. (colon)
    ❌ /Users/vagrant/git/Core/Core/API/APIError.swift:75:54: Colon Spacing Violation: Colons should be next to the identifier when specifying a type and next to the key in dictionary literals. (colon)
    /Users/vagrant/git/Core/Core/API/APIError.swift:76:32: error: Colon Spacing Violation: Colons should be next to the identifier when specifying a type and next to the key in dictionary literals. (colon)
    /Users/vagrant/git/Core/Core/API/APIError.swift:76:42: error: Colon Spacing Violation: Colons should be next to the identifier when specifying a type and next to the key in dictionary literals. (colon)
</details>
<details><summary>:x: XCTest failed: CoreTests/APIErrorTests/testCreateAccountErrors</summary>


    Assertion Failure at APIErrorTests.swift:50: XCTAssertEqual failed: ("") is not equal to ("code: invalid")
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Core/Core/Dashboard/K5/View/Grades/K5GradesView.swift [hunk 1]
```diff
@@ -107,13 +84,41 @@ struct K5GradesView: View {
         }
         .clipped()
     }
+
+    private var gradingPeriodSelectorState: some View {
```
Comment (idevainstr) [inline Core/Core/Dashboard/K5/View/Grades/K5GradesView.swift:88]: I'd rather call this gradingPeriodSelection, selectedGradingPeriod or similar.

**Review by idevainstr — DISMISSED** [review]

File: rn/Teacher/ios/Teacher/SpeedGrader/RubricAssessor.swift [hunk 2]
```diff
@@ -92,78 +92,126 @@ struct RubricAssessor: View {
...
+        let customGrade = (
+            assignment.freeFormCriterionCommentsOnRubric ||
+                assessment?.rating_id == nil ||
+                assessment?.rating_id == ""
+        )
+            ? assessment?.points : nil
+
+        return CircleToggle(
+            isOn:
+            Binding(
```
Comment (vargaat) [inline rn/Teacher/ios/Teacher/SpeedGrader/RubricAssessor.swift:138]: In such cases we could extract the Binding into a local variable (like `customGrade`) to increase readability.

**Review by vargaat — APPROVED** [review]

File: rn/Teacher/ios/Teacher/SpeedGrader/RubricAssessor.swift (hunk 2 above)
Comment (szabolcs-instructure) [reply rn/Teacher/ios/Teacher/SpeedGrader/RubricAssessor.swift:138]: My reason behind defining it there is that it's not used anywhere else but in this local scope only.

Comment (szabolcs-instructure) [reply rn/Teacher/ios/Teacher/SpeedGrader/RubricAssessor.swift]: I extracted the binding into its own variable!

**Review by vargaat — APPROVED** [review]

**Pull request merged**